proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+- Storage automatically provisioned via PVCs
+- Backup/restore follows standard Kubernetes PVC procedures
+
+#### Story 3: Operator Configuring Storage Class and Size
+
+As an operator, I want to configure KFP to use a specific StorageClass and PVC size instead of defaults, so that I can match storage performance and capacity to my workload requirements.
+
+**Acceptance Criteria:**
+
+- Can specify `StorageClass` in KFP configuration
+- Can set PVC size limits (global configuration)
+- Storage quotas enforced via Kubernetes `ResourceQuotas`
+- Clear error messages when storage limits are reached
+- Can choose between RWO and RWX access modes based on needs
+
+#### Story 4: User Migrating from S3 to Filesystem Storage
+
+As a user with existing pipelines containing components that call `boto3.upload_file()` directly, I want KFP system artifacts to use `kfp-artifacts://` with PVC storage while my custom components continue accessing S3, so that I can migrate incrementally without rewriting all components at once.
+
+**Acceptance Criteria:**
+
+- KFP artifacts use filesystem storage via `kfp-artifacts://` URIs
+- Components using boto3/gsutil for S3 continue to work
+- Can mix storage backends (system artifacts on PVC, custom data on S3)
+- Clear documentation on what uses filesystem vs object storage
+
+#### Story 5: Operator Deploying Multi-Tenant KFP with Namespace Isolation
+
+As an operator, I want to deploy KFP in namespace-local mode where each namespace annotated with `pipelines.kubeflow.org/enabled=true` gets its own artifact server pod and dedicated PVC, so that Team A's artifacts in namespace `team-a` are physically isolated from Team B's artifacts in namespace `team-b`.
+
+**Acceptance Criteria:**
+
+- Each namespace with `pipelines.kubeflow.org/enabled=true` annotation gets its own artifact server deployment
+- Each namespace gets its own dedicated PVC (no shared storage)
+- Artifact server in `team-a` namespace cannot access PVC in `team-b` namespace
+- Users can only access artifacts in namespaces they have RBAC permissions for (via `SubjectAccessReview`)
+- Physical isolation verified: deleting `team-a` namespace doesn't affect `team-b`'s artifacts
+
+#### Story 6: Operator Preferring KFP-Native Storage
+
+As an operator in a regulated environment (e.g., healthcare, finance), I want to deploy KFP with filesystem storage using an encrypted `StorageClass` (e.g., `encrypted-gp3`), so that artifact handling stays within the KFP codebase and I don't need to include a separate object storage system in my security audits.
+
+**Acceptance Criteria:**
+
+- All artifacts stored on PVCs within the cluster
+- KFP configuration uses `Filesystem.Type: "pvc"` with encrypted `StorageClass`
+- `SubjectAccessReview` validates all artifact access requests
+- Encryption at rest provided by the configured `StorageClass` (e.g., `encrypted-gp3`)
+- No separate object storage component to audit
```

Comment:
```suggestion
```

This story is also obsolete because you can just make the seaweedfs PVC encrypted.